modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/exec/UpdatableTableImpl.java

```diff
@@ -349,4 +357,25 @@ private static RuntimeException conflictKeysException(List<String> conflictKeys)
     private interface PartitionExtractor {
         int fromRow(BinaryRowEx row);
     }
+
+    private <RowT> void validateNotNullConstraint(RowHandler<RowT> rowHandler, List<RowT> rows) {
+        for (RowT row : rows) {
+            validateNotNullConstraint(rowHandler, row);
+        }
+    }
+
+    private <RowT> void validateNotNullConstraint(RowHandler<RowT> rowHandler, RowT row) {
+        for (int i = 0; i < desc.columnsCount(); i++) {
```

Comment:
Let's run benchmark, it looks a little bit not optimal.
Can we check which column is not nulable, keep the information and just after it start checking all required values for each rows?

Reply:
Also may be better to have new method for rowHandler to check value on null - it can be more optimal